mods/dungeon_loot/mapgen.lua

```diff
@@ -0,0 +1,159 @@
+minetest.set_gen_notify({dungeon=true})
+
+local function noise3d_integer(noise, pos)
+	return math.abs(math.floor(noise:get3d(pos) * 2147483647))
+end
+
+local function random_sample(rand, list, count)
+	local ret = {}
+	for n = 1, count do
+		local idx = rand:next(1, #list)
+		table.insert(ret, list[idx])
+		table.remove(list, idx)
+	end
+	return ret
+end
+
+local function find_walls(cpos)
+	local wall = minetest.registered_aliases["mapgen_cobble"]
+	local wall_alt = minetest.registered_aliases["mapgen_mossycobble"]
+	local wall_sand = minetest.registered_aliases["mapgen_sandstonebrick"]
+	local is_wall = function(node)
+		return table.indexof({wall, wall_alt, wall_sand}, node.name) ~= -1
+	end
+
+	local dirs = { {x=1, z=0}, {x=-1, z=0}, {x=0, z=1}, {x=0, z=-1} }
+	local get_node = minetest.get_node
+
+	local ret = {}
+	local mindist = {x=0, z=0}
+	local min = function(a, b) return a ~= 0 and math.min(a, b) or b end
+	local wallnode
+	for _, dir in ipairs(dirs) do
+		for i = 1, 8 do -- 8 = max room size / 2
+			local pos = vector.add(cpos, {x=dir.x*i, y=0, z=dir.z*i})
+
+			-- continue in that direction until we find a wall-like node
+			local node = get_node(pos)
+			if is_wall(node) then
+				local front_below = vector.subtract(pos, {x=dir.x, y=1, z=dir.z})
+				local above = vector.add(pos, {x=0, y=1, z=0})
+
+				-- check that it:
+				--- is at least 2 nodes high (not a staircase)
+				--- has a floor
+				if is_wall(get_node(front_below)) and is_wall(get_node(above)) then
+					table.insert(ret, {pos=pos, facing={x=-dir.x, y=0, z=-dir.z}})
+					if dir.z == 0 then
+						mindist.x = min(mindist.x, i-1)
+					else
+						mindist.z = min(mindist.z, i-1)
+					end
+					wallnode = node.name
+				end
+				-- abort even if it wasn't a wall cause something is in the way
+				break
+			end
+		end
+	end
+
+	return {
+		walls = ret,
+		size = {x=mindist.x*2, z=mindist.z*2},
+		type = wallnode == wall_sand and "desert" or "normal"
```

Comment:
Type "desert" should be for classic desert not sandstone desert.
We actually have 4 types of dungeon now:
Desert stone walls in biome "desert" (biome stone default:desert_stone).
Sandstone brick walls in biome "sandstone desert" (biome stone default:sandstone).
Cobble / mossycobble walls in many biomes (biome stone default:stone).
Then there is a recently added dynamic dungeon wall code, if the above 3 biome stone types are not found the dungeon wall node is set to the biome stone, so the 4th dungeon type is:
Ice walls in "icesheet" biome (biome stone default:ice).

Reply:
Maybe it was my mistake but i couldn't get the room detection code to detect the first type, is it possible that those are missing from gennotify?

I didn't even know that the last type existed, will add support for it.